### DISCUSSION TITLE
2.0.0-RC2

### DISCUSSION THREAD

**Discussion by vegegoku** [DominoKit/domino-ui#835] [Releases]
### Enhancements 🎉

* #802 LinkButton enhancements and use buttons for dialogs instead of LinkButtons.
* #804 Add the ability to set the default focus element when modal is opened.
* #806 Introduce the ability to override the ElementsFactory globally.
* #814 Hide dropdowns on 2nd click.
* #815 Allow read-only TextBox to look like a non-read-only.
* #821 Ability to register open/close listeners for side drawers(left | right).
* #824 Expose checkbox text element.
* #826 Hide dropdown when focus lost.
* #828 DateBox.closeOnBlur(true) auto closes the calendar popup when the user clicks away.
* #831 A DateBox set to readOnly shows the DatePicker.

### Fixes 🛠

* #796 ColumnFilter are not cleared when calling searchContext.clear.
* #799 ResizeColumnsPlugin in combination with SortPlugin triggers sorting when resizing a column.
* #800 SwitchButton & setReadOnly(true) color issue .
* #803 Calling clear or setValue(null) on DateBox throws a NPE.
* #807 TextAreaBox: clear() does not reset the letter counter.
* #809 FileUpload: setAutoUpload=true - the uploaded item shows an upload icon.
* #810 dui_bottom_0 is declared in two classes.
* #812 ColumnConfig: Uncaught DOMExceptions if the name of ColumnConfig contains special characters.
* #818 Setting background color for a datatable row in a plugin does not use the color variations.
* #820 Checkbox in a flex does not shrink to respect other elements flex grow.
* #823 Row is not deselected after selecting it twice using Datatable single SelectionPlugin.
* #827 HasCollapseListeners.removeCExpandListener method name typo.
* #832 Tree global selection listener not triggered when selecting TreeItems appended to the Tree root.
* #833 Bottom Elements are cut off.

**Full Changelog**: https://github.com/DominoKit/domino-ui/compare/1.0.0...2.0.0-RC2

<hr /><em>This discussion was created from the release <a href='https://github.com/DominoKit/domino-ui/releases/tag/2.0.0-RC2'>2.0.0-RC2</a>.</em>

**Comment by istockoapplitopia** [comment]
perfect guys
